dot/network/notifications.go

```diff
@@ -209,6 +214,59 @@ func (s *Service) createNotificationsMessageHandler(info *notificationsProtocol,
 	}
 }
 
+func (s *Service) sendData(peer peer.ID, hs Handshake, info *notificationsProtocol, msg NotificationsMessage) {
+	var err error
+	hsData, has := info.getHandshakeData(peer)
+	if !has || !hsData.received || !hsData.validated {
```

Comment:
if we've already received the data but it's not valid, then we should return (and probably close the stream with the peer, or disconnect from them)

Reply:
This is already done in the notification handler.

Reply:
in that case, can you remove the `|| !hsData.validated` case?

Reply:
Done